zebra-chain/src/sapling/shielded_data.rs

```diff
@@ -205,6 +206,15 @@ where
     AnchorV: AnchorVariant + Clone,
     Spend<PerSpendAnchor>: From<(Spend<AnchorV>, AnchorV::Shared)>,
 {
+    /// Iterate over the [`Spend`]s for this transaction, returning deduplicated
+    /// [`tree::Root`]s, regardless of the underlying transaction version.
+    pub fn anchors(&self) -> impl Iterator<Item = tree::Root> + '_ {
+        self.spends_per_anchor()
```

Comment:
Looks good, let's just leave ourselves a note to say how it could be improved.

```suggestion
        // TODO: use TransferData::shared_anchor to improve performance for V5 transactions
        self.spends_per_anchor()
```